smartcontractkit/chainlink
# release/1.7.0 -> master

- Additional errored states for StartStopOnce (#6887)
- Bump github/codeql-action from 2.1.14 to 2.1.15 (#6888)
- Bump github.com/gin-gonic/gin from 1.7.7 to 1.8.1 (#6769)
- Bump github.com/tidwall/gjson from 1.9.3 to 1.14.1 (#6518)
- implement optimism validator contracts (#6885)
- VRF v2: add max gas price gwei in vrf job spec (#6867)
- bump spf13/cobra to v1.5.0 (#6894)
- EVM basic chaos suite (#6889)
- fix changelog (#6897)
- Better errors on user creation (#6896)
- feat: DKG encryption keystore support (#6825)
- `gasLimit` job attribute (#6865)
- Adds Soak Tests, Cleans Up Smoke Tests (#6898)
- core/services/chainlink: config dump test cases w/ db (#6893)
- core/services/chainlink: config dump test cases w/ db (#6893)
- Implemented integration test focused on the migration between registries of upkeeps (#6892)
- bump deps to fix nodes < tests failures (#6909)
- Improve handling of missing blocks (#6904)
- Add test to fwdrmgr and remove ocrv1 auth checks (#6907)
- bump libocr (#6913)
- .github/workflows: CI Core: schedule nightly run w/ increased race test timeout (#6922)
- core/chains/evm/client: log web3_clientVersion at debug instead of trace (#6929)
- .github/workflows: bump setup-go to v3 for automatic version recognition and go mod caching (#6932)
- ci: fix lint (#6934)
- core/internal/testutils: use test logger instead of log.Print* (#6930)
- chore: vrf debugging commands (#6928)
- Clean out account number usage and add code owners
- VRFV2 Wrapper (#6940)
- Clearer error messaging for unknown RPC errors (#6946)
- Do not log DB urls on insufficient password complexity (#6949)
- Exponential backoff retry on EthBroadcaster send transaction timeout (#6948)
- remove inconsistent comments about getRoundData
- Add official support for Besu execution client (#6947)
- OCR2VRF DKG Plugin (#6849)
- Refactored Keeper Integration Tests to Use Table tests (#6912)
- Update build-custom workflow for OIDC role only (#6943)
- core/chains/evm/txmgr: Test_EthResender increase eventually interval (#6974)
- update CLI with turn taking algo (#6956)
- Use pending nonce to confirm if an errored transaction actually succeeded (#6964)
- implementation for ocr2vrf JuelsForFeeCoin (#6971)
- Squashing all changes to make the signed commit (#6795)
- core/internal/testutils: resolve test log race in newWSHandler (#6986)
- Add Sol verified cached client wrapper (#6920)
- implement blockhashes interface for ocr2vrf (#6965)
- Add OCR2VRF contract wrapper script for generation inside Chainlink repo. (#6991)
- Added back the table tests which were overwritten by another PR (#6993)
- chore: remove dkg skeleton (#6994)
- Remove build-custom in favor of using the action in chainlink-github-actions https://github.com/smartcontractkit/chainlink-github-actions/blob/main/chainlink-testing-framework/build-image/action.yml
- Change compile script to use 8.15 compiler for VRFV2Wrapper contracts, to match gauntlet. (#6989)
- Update passwords, fix build_abigen, go-ethereum 1.10.20 (#6969)
- update keeper cli to work with registry 1.2 (#6977)
- reorg for the runlog job (#6937)
- per job-type gas limit (#6935)
- Fixed flakey tests in head_tracker (#7000)
- Remove myself from some CODOWNERS entries (#6999)
- mockery: bump and regenerate with 2.14.0 (#7004)
- Add report serializer, leave deserializer unimplemented. (#6996)
- Take into account db URL params when validating password (#6963)
- Enable Solana E2E Smoke Tests Again (#6988)
- feat: ocr2vrf CoordinatorInterface implementation (#6981)
- add CLI commands for ocr2vrf deployment, configuration and request (#6998)
- Fix a bug in dump-config command (#6968)
- Soak Test Overhaul (#6997)
- Setup paths and a dummy integration-test workflow so we can make integration tests a required check (#7007)
- feat: OCR2VRF launch codes (#7003)
- increase memory for solidity coverage action (#6984)
- Contracts release 0.4.2 (#6972)
- bump chainlink-env lib
- Added `hexdecode` and `base64decode` tasks (#7014)
- Switch ocr2vrf wrappers to chainlink internal for coordinator (#7017)
- Retry indefinitely on TxFeeExceedsCapError (#7019)
- Multi-user and RBAC support (#6843)
- Setup KeeperRegistry for development (#7027)
- For VRFV2 superscript, optionally deploy LINK token + LINK/ETH feed if not provided, and fund sending oracle automatically. (#7005)
- Avoid running linting CI check on PRs from release branches (#7028)
- Avoid running linting CI check on PRs from release branches (#7028)
- pipeline tests cleanup (#7026)
- Just testing (#7030)
- Updates config test
- Story/39608 refactor (#7033)
- Chore/beacon generate gethwrappers for beacon consumer and update imports (#7011)
- Integration tests which takes down half of the Keeper Nodes (#6992)
- Update to run with auto-network switching (#7038)
- increase timeout for verifying upkeep performs in integration test (#7049)
- Fix publish test report (#7041)
- Fixed mockery usage for FM (#7047)
- Fixed mockery usage for pipeline (#7048)
- split keeper registry into proxy and logic (#7016)
- internal/config: fix TOML parse error (#7051)
- Adjust ocr2vrf wrapper and compile scripts to reduce optimization runs. (#7044)
- Soak fix (#7052)
- Move SecretGenerator out of Config (#7046)
- Klaytn support (#7053)
- Upgrade solc wrapper compiler version to 8.15 for VRFV2Wrappers (#7054)
- Adds Goerli Support to run tests (#7058)
- feat: automate ocr2vrf job creation (#7018)
- Fixing starknet dependencies (#7069)
- fix: multiple transmissions from same oracle (#7068)
- Move admin functions from proxy to logic (#7064)
- Remove email from password message (#7076)
- bump libocr for Klaytn support (#7077)
- Add official support for Sepolia chain (#7078)
- Assign the existing single API user as admin when migrating up to RBAC feature (#7080)
- Add support for besu's non-unique logIndex (#7072)
- Chore/update gethwrappers 7 21 (#7071)
- Fixed job stuck in suspended state due to failed task (#7074)
- Log poller internal review (#7067)
- Fixes a bug causing an FMS job proposal to not be approved. (#7082)
- fix: vrf transmission from all ocr nodes (#7081)
- Wrap api calls with go-resty library (#7073)
- Update go.mod (#7086)
- Revert example.env changes (#7088)
- Reshuffle erroneous changelog entries from 1.6 to unreleased
- Bump version and update CHANGELOG for core v1.7.0
- Fix logging of msg in ethereum requests
- Update log/pool.go & log/pool_test.go for besu (#7106)
- Convert NULL tx_index values in log_broadcaster to -1
- Fix bug in getting highest confirmed nonce
- DROP tx_index column from log_broadcasts table
- Revert "Update log/pool.go & log/pool_test.go for besu (#7106)"
- Revert "Add support for besu's non-unique logIndex (#7072)"
- Minor fixes
- Finalize date on changelog for 1.7.0
